### PR TITLE
Enable support for adding R Script to Pipeline

### LINKED ISSUE #1424: Enable adding R Scripts to pipelines
**Describe the issue**

Similar to how we now support Python Scripts as pipeline nodes, R scripts are also very popular and should be supported in the pipeline editor.

**Tasks**

- [x] Enable drag and drop of R scripts to pipeline editor
- [x] Enable running pipelines in KFP, Airflow, and local mode
- [x] Add/Install sample R runtime-image 

**Pipeline Editor example**

image

### PR DESCRIPTION
**Pull request by lresende**
Fixes #1424 

Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by kevin-bates** [comment]
I expected to see support for local pipeline runs.  The changes should be on par with what took place in kfp-notebook.  Any chance those could be included in this PR?

File: packages/pipeline-editor/src/canvas.ts [hunk 1]
```diff
@@ -19,23 +19,31 @@ import { IconUtil } from '@elyra/ui-components';
...
+  pythonIcon,
+  rKernelIcon
+} from '@jupyterlab/ui-components';
 
 import { PipelineService } from './PipelineService';
 
 enum ContentType {
   notebook = 'notebook',
   python = 'python',
+  r = 'r',
```
Comment (ajbozarth) [inline packages/pipeline-editor/src/canvas.ts:34]: I think it may be better to call this `r_script` or `rScript` instead of having a one char name

**Comment by lresende** [comment]
> I expected to see support for local pipeline runs. The changes should be on par with what took place in kfp-notebook. Any chance those could be included in this PR?

Figuring out some issues related to installing R on mac os via conda, and should be able to push the updates to enable local runs.

**Comment by kevin-bates** [comment]
Cool. Thanks for dealing with that detail!

**Comment by lresende** [comment]
Updated the local processor with R script support:

```
[D 2021-03-11 21:14:38.765 ServerApp] Adding operation for 'sample-python/factorial.py' to pipeline: factorial
[D 2021-03-11 21:14:38.765 ServerApp] Adding operation for 'sample-python/factorial.r' to pipeline: factorial
[I 2021-03-11 21:14:38.765 ServerApp] local 'factorial' - processing pipeline 
[D 2021-03-11 21:14:38.765 ServerApp] Processing R script: /Users/lresende/opensource/jupyter-notebooks/sample-python/factorial.r
\[1] "The factorial of 7 is 5040"
[D 2021-03-11 21:14:39.109 ServerApp] Execution of factorial.r took 0.343 secs.
[I 2021-03-11 21:14:39.109 ServerApp] local 'factorial':'factorial' - completed sample-python/factorial.r (0.344 secs)
[D 2021-03-11 21:14:39.109 ServerApp] Processing python script: /Users/lresende/opensource/jupyter-notebooks/sample-python/factorial.py
|The factorial of 7 is 5040
[D 2021-03-11 21:14:39.154 ServerApp] Execution of factorial.py took 0.044 secs.
[I 2021-03-11 21:14:39.154 ServerApp] local 'factorial':'factorial' - completed sample-python/factorial.py (0.045 secs)
[I 2021-03-11 21:14:39.154 ServerApp] local 'factorial' - pipeline processed (0.389 secs)
```

**Comment by lresende** [comment]
Installing R locally with conda, you can add the necessary packages when creating the `dev` conda environment:
```
conda create -y --name dev jupyterlab pandas numpy r r-essentials r-irkernel
```

Having said that, I was trying to add these to `clean-jupyterlab.sh` and was having some issues which I will still take a deeper look to see if there is a way to avoid the conflicts.

**Review by kevin-bates — APPROVED** [review]
LGTM - we can look at refactoring script-based processors later (if necessary).

Thanks for putting this in so quickly!

**Comment by ptitzler** [comment]
We need to also predefine an "experimental" container image with R preinstalled, publish it, and create a default runtime image configuration. Without one, users would first have to create one themselves before they can even try this out on KFP or AA. (Think of this as including a charged battery with an electronic device. Users are much happier if they can just turn the device on and use it right away.)

**Comment by lresende** [comment]
> We need to also predefine an "experimental" container image with R preinstalled, publish it, and create a default runtime image configuration. Without one, users would first have to create one themselves before they can even try this out on KFP or AA. (Think of this as including a charged battery with an electronic device. Users are much happier if they can just turn the device on and use it right away.)

It's already in https://github.com/elyra-ai/elyra/pull/1418/commits/8f2cbb2b6b0813ed35ab54c3f7ebe900efa32bbb

**Comment by ptitzler** [comment]
> It's already in [8f2cbb2](https://github.com/elyra-ai/elyra/commit/8f2cbb2b6b0813ed35ab54c3f7ebe900efa32bbb)

Ah yes. There was nothing in the PR description and I haven't gone through a code review yet. Therefore I missed that.

**Comment by ptitzler** [comment]
We need to bump the pipeline version number to prevent failures should one try to open a pipeline file using an older version of Elyra, no?

**Comment by ajbozarth** [comment]
I'm not a huge fan of the R icon from lab so I tried out replacing it in lab and replacing it in pipelines (couldn't do both at once) with the official logo.

Screen Shot 2021-03-15 at 3 02 19 PM
Screen Shot 2021-03-15 at 3 17 25 PM

I'm ok merging as is, but wanted some discussion on:

1. Should we use the official logo for pipeline, even if it's different from lab?
2. Should we open a PR updating the logo in lab?

**Comment by lresende** [comment]
> I'm ok merging as is, but wanted some discussion on:
> 
> 1. Should we use the official logo for pipeline, even if it's different from lab?

I am in favor of reusing what is available in lab for consistency. 

> 2. Should we open a PR updating the logo in lab?

Up to you, as long as we are consistent, I am ok.

**Comment by ajbozarth** [comment]
So I did some digging and found the origin of the R icon in lab. It was added in this PR https://github.com/jupyterlab/jupyterlab/pull/2558
It seems it was an icon specifically designed for use in JupyterLab, specifically the file browser and launcher. 
I will open an issue on lab to see if they're open to changing it, I'm worried that using a jupyterlab specific icon, that is significantly different from the official icon, in pipelines will be confusing when someone uses the vscode pipeline extension and comes across the logo which wouldn't match the one used in vscode's UI

**Comment by ajbozarth** [comment]
actually on further consideration, there is already precedence for different icons to be used in the launcher than in the file browser (python logo). I would argue that we should use the official R Logo in Elyra code, both here in pipelines, but also when we add R script support and add a launcher icon for that

**Pull request merged**